presidio-recognizers-store/cmd/presidio-recognizers-store/main.go

```diff
@@ -298,10 +298,16 @@ func applyGetHash() (*types.RecognizerHashResponse, error) {
 	}
 
 	hash, err := recognizersStore.Get(hashKey)
-	if err != nil || hash == "" {
+	if err != nil {
 		errMsg := "Failed to find the latest hash"
 		log.Error(errMsg)
-		return &types.RecognizerHashResponse{}, errors.New(errMsg)
+		return nil, errors.New(errMsg)
+	}
+
+	// no error, however hash was not found, this means the redis is still
+	// empty.
+	if hash == "" {
```

Comment:
shouldn't this be 
if not hash {

Reply:
Short answer is 'no'.
long answer is that you are correct. redis returns redis.Nil in the error object if item is not found.
however in our implementation ([Line 48-49](https://github.com/Microsoft/presidio/blob/development/pkg/cache/redis/redis.go#L48)) we return "" in such case (our redis wrapper)
so thats why my check is correct